src/components/frame/v5/pages/ColonyPreviewPage/ColonyPreviewPage.tsx

```diff
@@ -0,0 +1,256 @@
+  },
+  description: {
+    id: `${displayName}.description`,
+    defaultMessage:
+      'The Colony app is in private beta, allowing invited members and Colony’s to test out the new features before launch.',
+  },
+  connectWalletTitle: {
+    id: `${displayName}.connectWalletTitle`,
+    defaultMessage: 'Connect your wallet to check your access',
+  },
+  connectWalletSubtitle: {
+    id: `${displayName}.connectWalletSubtitle`,
+    defaultMessage: 'You might have private beta access already available.',
+  },
+  notificationBannerTitle: {
+    id: `${displayName}.notificationBannerTitle`,
+    defaultMessage:
+      'You have been invited to join {colonyName} for the private beta.',
+  },
+  invalidBannerTitle: {
+    id: `${displayName}.invalidBannerTitle`,
+    defaultMessage: 'Sorry, your invite code is not valid. Please check again.',
+  },
+  connectWalletButton: {
+    id: `${displayName}.connectWalletButton`,
+    defaultMessage: 'Connect wallet',
+  },
+  privateAccessHeading: {
+    id: `${displayName}.privateAccessHeading`,
+    defaultMessage: 'Private access only',
+  },
+  restrictedAccessMessage: {
+    id: `${displayName}.restrictedAccessMessage`,
+    /* eslint-disable max-len */
+    defaultMessage: `
+    This Colony has restricted access during the private beta test. Only members who have been invited can access the Colony.
+      {needsToRequestAccess, select,
+      true {To request access to this Colony, you can contact them via their social accounts above.}
+      other {}
+    }`,
+    /* eslint-enable max-len */
+  },
+  joinColonyButton: {
+    id: `${displayName}.joinColonyButton`,
+    defaultMessage: 'Join the Colony',
+  },
+});
+
+const ColonyPreviewPage = () => {
+  const { inviteCode = '', colonyName = '' } = useParams<{
+    inviteCode: string;
+    colonyName: string;
+  }>();
+  const { pathname } = useLocation();
+  const { formatMessage } = useIntl();
+  const { connectWallet, wallet, user, userLoading, walletConnecting } =
+    useAppContext();
+
+  const navigate = useNavigate();
+
+  const { data: inviteData, loading: inviteLoading } =
+    useGetColonyMemberInviteQuery({
+      variables: { id: inviteCode },
+      skip: !inviteCode,
+    });
+
+  // @TODO: This is terrible. Once we have auth, we need a method
+  // to check whether the logged in user is a member of the Colony
+  const { data: whitelistData, loading: whitelistLoading } =
+    useGetColonyWhitelistByNameQuery({
+      variables: { name: colonyName },
+    });
+
+  const { data: colonyData, loading: colonyLoading } =
+    useGetPublicColonyByNameQuery({
+      variables: { name: colonyName },
+      skip: !colonyName,
+    });
+
+  const [validate] = useValidateUserInviteMutation();
+
+  const colonyAddress = colonyData?.getColonyByName?.items[0]?.colonyAddress;
+  const colonyMetadata = colonyData?.getColonyByName?.items[0]?.metadata;
+
+  const validateInviteCode = useCallback(async () => {
```

Comment:
`useCallback`s purpose is to prevent the creation of new functions on the "stack" when a component is re-rendered. How this function is used should be irrelevant.

Reply:
But there weren't any performance issues caused by that, were there? This just sounds like a premature optimisation and it's not a use case mentioned in the react docs: https://react.dev/reference/react/useCallback